components/forms/StageForm.tsx

```diff
@@ -0,0 +1,92 @@
+import { Button, DatePicker, Form, Input, Select } from 'antd';
+import { stageTypeToDisplayStringMap } from '../../utils/applicationStage/applicationStageUtils';
+import AddEmojiIcon from '../icons/AddEmojiIcon';
+import { Dispatch, MouseEvent, SetStateAction, useEffect, useState } from 'react';
+import dynamic from 'next/dynamic';
+import { IEmojiData } from 'emoji-picker-react';
+import { Nullable } from '../../types/utils';
+import { ApplicationStageFormData } from '../../types/applicationStage';
+
+interface Props {
+  initialValues: ApplicationStageFormData;
+  isSubmitting: boolean;
+  setIsSubmitting: Dispatch<SetStateAction<boolean>>;
+  setStageFormData: Dispatch<SetStateAction<Nullable<ApplicationStageFormData>>>;
+}
+
+function StageForm({ initialValues, isSubmitting, setIsSubmitting, setStageFormData }: Props) {
+  const [form] = Form.useForm();
+
+  const [shouldShowEmojiPicker, setShouldShowEmojiPicker] = useState<boolean>(false);
+  const [selectedEmojiUnicode, setSelectedEmojiUnicode] = useState<Nullable<string>>(null);
+
+  const DynamicPicker = dynamic(() => import('emoji-picker-react'));
+
+  const onSelectEmoji = (event: MouseEvent, emojiObject: IEmojiData) => {
+    setSelectedEmojiUnicode(emojiObject.unified);
+    setShouldShowEmojiPicker(false);
+  };
+
+  useEffect(() => {
+    form.resetFields();
+
+    if (initialValues.emojiUnicodeHex !== undefined) {
+      setSelectedEmojiUnicode(initialValues.emojiUnicodeHex);
+    }
+  }, [initialValues]);
+
+  useEffect(() => {
+    if (isSubmitting) {
+      form
+        .validateFields()
+        .then(() => setStageFormData({ ...form.getFieldsValue(), emojiUnicodeHex: selectedEmojiUnicode }))
+        .catch(() => setIsSubmitting(false));
+    }
+  }, [isSubmitting]);
+
+  return (
+    <Form form={form} initialValues={initialValues}>
+      <Form.Item name="type" label="Stage" rules={[{ required: true, message: 'Please choose a stage.' }]}>
+        <Select placeholder="Select stage">
+          {stageTypeToDisplayStringMap.map((value, key) => (
```

Comment:
The order of the stages right now is random, because of the `Immutable.Map`, which can make it quite confusing for the user.

Should we use a [Immutable.OrderedMap](https://immutable-js.com/docs/v4.1.0/OrderedMap/) for `stageTypeToDisplayStringMap` instead?